src/ethernet/RsRtspClient.cpp

```diff
@@ -0,0 +1,613 @@
+  return key;
+}
+
+//TODO: change char* to std::string
+IRsRtsp *RsRTSPClient::getRtspClient(char const *t_rtspURL,
+                                     char const *t_applicationName, portNumBits t_tunnelOverHTTPPortNum)
+{
+  TaskScheduler *scheduler = BasicTaskScheduler::createNew();
+  UsageEnvironment *env = BasicUsageEnvironment::createNew(*scheduler);
+  
+  RTSPClient::responseBufferSize = 100000;
+  return (IRsRtsp *)new RsRTSPClient(scheduler,env, t_rtspURL, RTSP_CLIENT_VERBOSITY_LEVEL, t_applicationName, t_tunnelOverHTTPPortNum);
+  //return rtspClient;
+}
+
+RsRTSPClient::RsRTSPClient(TaskScheduler *t_scheduler,UsageEnvironment *t_env, char const *t_rtspURL,
+                           int t_verbosityLevel, char const *t_applicationName, portNumBits t_tunnelOverHTTPPortNum)
+    : RTSPClient(*t_env, t_rtspURL, t_verbosityLevel, t_applicationName, t_tunnelOverHTTPPortNum, -1)
+{
+  m_lastReturnValue.exit_code=RsRtspReturnCode::OK;
+  m_env = t_env;
+  m_scheduler = t_scheduler;
+}
+
+RsRTSPClient::~RsRTSPClient()
+{
+}
+
+std::vector<rs2_video_stream> RsRTSPClient::getStreams()
+{
+  // TODO - handle in a function
+  unsigned res = this->sendDescribeCommand(this->continueAfterDESCRIBE);
+  if (res == 0)
+  {
+    // An error occurred (continueAfterDESCRIBE was already called)
+    // TODO: return error code
+    return this->m_supportedProfiles;
+  }
+  // wait for continueAfterDESCRIBE to finish
+  std::unique_lock<std::mutex> lck(m_commandMtx);
+  m_cv.wait_for(lck, std::chrono::seconds(RTSP_CLIENT_COMMANDS_TIMEOUT_SEC), [this] { return m_commandDone; });
+  // for the next command - if not done - throw timeout
+  if(!m_commandDone)
+  {
+    RsRtspReturnValue err = 
+      {RsRtspReturnCode::ERROR_TIME_OUT,"client time out"};
+    throw std::runtime_error(format_error_msg(__FUNCTION__, err));
+  }
+  m_commandDone = false;
+  
+  if (m_lastReturnValue.exit_code!=RsRtspReturnCode::OK)
+  {
+    throw std::runtime_error(format_error_msg(__FUNCTION__, m_lastReturnValue));
+  }
+
+  if (this->m_supportedProfiles.size()==0)
+  {
+    RsRtspReturnValue err = 
+      {RsRtspReturnCode::ERROR_GENERAL, std::string("failed to get streams from network device at url: " + std::string(this->url()))};
+    throw std::runtime_error(format_error_msg(__FUNCTION__, err));
+  }
+
+  return this->m_supportedProfiles;
+}
+
+int RsRTSPClient::addStream(rs2_video_stream t_stream, rtp_callback *t_callbackObj)
+{
+  long long int uniqueKey = getStreamProfileUniqueKey(t_stream);
+  RsMediaSubsession *subsession = this->m_subsessionMap.find(uniqueKey)->second;
+  if(subsession == nullptr)
+  {
+    RsRtspReturnValue err = 
+      {RsRtspReturnCode::ERROR_WRONG_FLOW,"requested stream was not found"};
+    throw std::runtime_error(format_error_msg(__FUNCTION__, err));
+  }
+  
+    if (!subsession->initiate())
+    {
+      this->envir() << "Failed to initiate the subsession \n";
```

Comment:
Is this a log ? Can you you LRS infrastructure here?

Reply:
Done